src/lib/flow/include/sol-flow-parser.h

```diff
@@ -72,11 +73,10 @@ struct sol_flow_parser_client {
      * @param data Client's data
      * @param name File name to read
      * @param buf Where to store the file
-     * @param size Buffer's size
      *
      * @return @c 0 on success, error code (always negative) otherwise
      */
-    int (*read_file)(void *data, const char *name, const char **buf, size_t *size);
+    int (*read_file)(void *data, const char *name, struct sol_buffer *buf);
```

Comment:
Explicitly say that the buffer must be already initialized and data will be appended. If the data does not fit, an error is returned